src/main/scala/org/deeplearning4j/scalphagozero/scoring/GameResult.scala

```diff
@@ -53,64 +96,62 @@ object GameResult {
     * @param goBoard GoBoard instance
     * @return Territory object
     */
-  def evaluateTerritory(goBoard: GoBoard): Territory = {
-    val statusMap = new mutable.HashMap[Point, String]()
-    for (row <- 1 to goBoard.row) {
-      for (col <- 1 to goBoard.col) {
-        val point = Point(row, col)
-        if (!statusMap.contains(point)) {
-          val stoneColor: Option[Player] = goBoard.getPlayer(point)
-          if (stoneColor.isDefined) {
-            val color = stoneColor.get
-            val status = if (color == BlackPlayer) "black" else "white"
-            statusMap.put(point, status)
-          } else {
+  def evaluateTerritory(goBoard: GoBoard): Map[Point, GamePointType] = {
+    val statusMap = mutable.Map.empty[Point, GamePointType]
+    for (row <- 1 to goBoard.row; col <- 1 to goBoard.col) {
+      val point = Point(row, col)
+      if (!statusMap.contains(point)) {
+        goBoard.getPlayer(point) match {
+          case Some(color) =>
+            statusMap.put(point, if (color == BlackPlayer) BlackStone else WhiteStone)
+          case None =>
             val (group, neighbors) = collectRegion(point, goBoard)
-            var fillWith: String = ""
-            if (neighbors.size == 1) {
-              val neighborColor: Option[Player] = neighbors.head
-              val stoneString = if (neighborColor.get == BlackPlayer) "b" else "w"
-              fillWith = "territory_" + stoneString
-            } else {
-              fillWith = "dame"
-            }
+            val fillWith =
+              if (neighbors.size == 1) {
+                val neighborColor: Option[Player] = neighbors.head
+                if (neighborColor.get == BlackPlayer) BlackTerritory else WhiteTerritory
+              } else {
+                Dame
+              }
             for (position <- group) {
               statusMap.put(position, fillWith)
             }
-          }
         }
       }
     }
-    new Territory(statusMap)
+    statusMap.toMap
   }
 
+  private[this] final val deltas = List((-1, 0), (1, 0), (0, -1), (0, 1))
```

Comment:
Don't loose time to review the code following this line. 

@sderosiaux has a PR for you simplifying the `collectRegion` function.